File: solve.cpp
#include "main.h"

void RunConverter(std::istream& is) {
    std::string task;
    while (std::getline(is, task)) {
        Solve(task);
    }
}

void Solve(std::string& task) {
    std::vector<Node> parsedExpr;
    task.erase(remove(task.begin(), task.end(), ' '), task.end());
     if (Parse(task, parsedExpr)) {
        int idx = 0;
        double answer = CalculateExpr(idx, parsedExpr);
        std::cout << ConvertToRoman(answer) << std::endl;
    }
    else {
        std::cout << "error: wrong input format" << std::endl;
    }
}

File: go/solve.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"strings"
)

func RunConverter(reader io.Reader) {
	scanner := bufio.NewScanner(reader)
	for scanner.Scan() {
		Solve(scanner.Text())
	}
}

func Solve(task string) {
	parsedExpr := make([]Node, 0)
	task = strings.ReplaceAll(task, " ", "")
	if Parse([]rune(task), &parsedExpr) {
		idx := 0
		answer := CalculateExpr(&idx, parsedExpr)
		fmt.Println(ConvertToRoman(answer))
	} else {
		fmt.Println("error: wrong input format")
	}
}
